fil_fungible_token/src/receiver/types.rs

```diff
@@ -12,11 +13,14 @@ pub trait FRC46TokenReceiver {
     /// the receiving actor can immediately utilise the received funds. If the receiver wishes to
     /// reject the incoming transfer, this function should abort which will cause the token actor
     /// to rollback the transaction.
-    fn tokens_received(params: TokensReceivedParams);
+    fn receive(token_type: TokenType, params: FRC46TokenReceived);
```

Comment:
Perhaps generalise this to be a UniversalReceiver trait, and with the params argument here being raw bytes. We're going to want essentially the same thing for NFT (so perhaps this will move out of this crate to a new fil_universal_receiver crate too). 

But you could convince me that being generic for all receivers is out of scope here.

Reply:
Moving things around can wait for a follow-up of course

Reply:
Updated to this: 
```Rust
fn receive(params: UniversalReceiverParams);
```

I've also added more comments on it and the `UniversalReceiverParams` which hopefully explain things clearly